### ISSUE TITLE
Remove deprecated twisted.python.hashlib

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7905]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#7905|
|Type|task|
|Created|2015-05-19 23:42:23Z|
|Branch|https://github.com/twisted/twisted/tree/remove-hashlib-7905|

[twisted/python/hashlib.py](twisted/python/hashlib.py) was deprecated in Twisted 13.1 ([#6342](https://github.com/twisted/twisted/issues/6342)) and can be removed.

<details><summary>Searchable metadata</summary>

```
trac-id__7905 7905
type__task task
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_remove_hashlib_7905 branches/remove-hashlib-7905
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1432078943114846 1432078943114846
changetime__1432276687201288 1432276687201288
version__None None
owner__hawkowl hawkowl
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [44889]) Branching to remove-hashlib-7905.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Changes made, builders spun, please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

look good to merge.

not all builders are ready but I assume the remaining one will also pass.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [44906]) Merge remove-hashlib-7905: Remove the deprecated twisted.python.hashlib

Author: hawkowl
Reviewer: adiroiban
Fixes: [#7905](https://github.com/twisted/twisted/issues/7905)
